lib/spack/env/cc

```diff
@@ -108,6 +108,12 @@ case "$command" in
         ;;
     ld)
         mode=ld
+        for arg in "$@"; do
+            if [ "$arg" = -S ]; then
```

Comment:
This catches the case where `ld` is called with an `-S` option. I don't think this should happen -- the `-S` option is for the compiler driver `cc`.

There apparently is an `-S` option for `ld`; it means the opposite of `-g`, i.e. remove (Strip) all debug information. This has nothing to do with the mode (the mode should still be `ld`), nor with how `-rpath` should be handled.